.github/workflows/Deploy.yaml

```diff
@@ -27,49 +31,104 @@ on:
+      copyToMain: ${{ steps.CreateInputs.outputs.copyToMain }}
+      directCommit: ${{ steps.CreateInputs.outputs.directCommit }}
+      requireEndToEndTests: ${{ steps.CreateInputs.outputs.requireEndToEndTests }}
+      createRelease: ${{ steps.CreateInputs.outputs.createRelease }}
+      defaultBcContainerHelperVersion: ${{ steps.CreateInputs.outputs.defaultBcContainerHelperVersion }}
+    steps:
+      - name: Create inputs
+        id: CreateInputs
+        run: |
+          if ($env:GITHUB_EVENT_NAME -eq 'workflow_run') {
+            $branch = 'preview'
+            $copyToMain = 'false'
+            $directCommit = 'false'
+            $requireEndToEndTests = 'true'
+            $createRelease = 'false'
+            $defaultBcContainerHelperVersion = 'preview'
+          } else {
+            $branch = '${{ github.event.inputs.branch }}'
+            $copyToMain = '${{ github.event.inputs.copyToMain }}'
+            $directCommit = '${{ github.event.inputs.directCommit }}'
+            $requireEndToEndTests = '${{ github.event.inputs.requireEndToEndTests }}'
+            $createRelease = '${{ github.event.inputs.createRelease }}'
+            $defaultBcContainerHelperVersion = '${{ github.event.inputs.defaultBcContainerHelperVersion }}'
+          }
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "branch=$branch"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "copyToMain=$copyToMain"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "directCommit=$directCommit"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "requireEndToEndTests=$requireEndToEndTests"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "createRelease=$createRelease"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "defaultBcContainerHelperVersion=$defaultBcContainerHelperVersion"
+
   CheckEndToEnd:
     runs-on: [ ubuntu-latest ]
+    needs: [ Inputs ]
     steps:
       - name: Check successful end 2 end tests have run
-        if: github.repository_owner == 'microsoft' && github.event.inputs.requireEndToEndTests == 'true'
+        if: github.repository_owner == 'microsoft' && needs.Inputs.outputs.requireEndToEndTests == 'true'
         env:
           GH_TOKEN: ${{ github.token }}
         run: |
           $errorActionPreference = "Stop"
-          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/runs?per_page=100 | ConvertFrom-Json).workflow_runs | Where-Object { $_.name -eq 'End 2 End Tests' }
+          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/workflows/E2E.yaml/runs?per_page=100 | ConvertFrom-Json).workflow_runs
           $latestSha = (gh api /repos/$($env:GITHUB_REPOSITORY)/commits/main | ConvertFrom-Json).sha
-          $latestRun = $end2endruns | Where-Object { $_.head_sha -eq $latestSha }
+          $latestRun = $end2endruns | Where-Object { $_.head_sha -eq $latestSha } | select-object -first 1
```

Comment:
The API has `status` and `head_sha` parameter that could be utilized in this case. See https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow


I think it's best to use these parameters as latest run might not be successful, but the previous run (on the same SHA) might have been successful. It's weird case (why would we run e2e tests on the same SHA if the previous run succeeded), but who knows... we could reach the point where we run e2e a couple of times in hopes that one of them is fully successful.

Reply:
With the other discussion in place, the `head_sha` won't be needed, but the status would be useful.